src/LinkDotNet.NCronJob/Configuration/DynamicJobFactory.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.Extensions.DependencyInjection;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace LinkDotNet.NCronJob;
+
+internal class DynamicJobFactory : IJob
+{
+    private readonly IServiceProvider serviceProvider;
+    private readonly Func<object[], Task> invoker;
+    private readonly Func<IServiceProvider, object>?[] serviceResolvers;
+    private readonly ParameterInfo[] parameters;
+
+    public DynamicJobFactory(IServiceProvider serviceProvider, Delegate jobAction)
+    {
+        ArgumentNullException.ThrowIfNull(jobAction);
+
+        this.serviceProvider = serviceProvider;
+        parameters = jobAction.Method.GetParameters();
+        serviceResolvers = BuildServiceResolvers(parameters);
+        invoker = BuildInvoker(jobAction);
+    }
+
+    private static Func<object[], Task> BuildInvoker(Delegate jobAction)
+    {
+        var method = jobAction.Method;
+        var returnType = method.ReturnType;
+        var param = Expression.Parameter(typeof(object[]), "args");
+        var args = method.GetParameters().Select((p, index) =>
+            Expression.Convert(Expression.ArrayIndex(param, Expression.Constant(index)), p.ParameterType)).ToArray();
+        var call = Expression.Call(Expression.Constant(jobAction.Target), method, args);
+
+        if (returnType == typeof(Task))
+        {
+            var lambda = Expression.Lambda<Func<object[], Task>>(call, param);
+            return lambda.Compile();
+        }
+        else if (returnType == typeof(void))
+        {
+            var lambda = Expression.Lambda<Action<object[]>>(Expression.Block(call, Expression.Default(typeof(void))), param);
+            var action = lambda.Compile();
+            return objects => { action(objects); return Task.CompletedTask; };
+        }
+        else
+        {
+            throw new InvalidOperationException("The job action must return a Task or void.");
+        }
+    }
+
+
+    private static Func<IServiceProvider, object>?[] BuildServiceResolvers(ParameterInfo[] parameters) =>
+        parameters.Select(p =>
```

Comment:
I would prefer the whole lambda as a local function. Mainly because the else branch does quite a lot